api/src/database/index.ts

```diff
@@ -19,6 +19,13 @@ export default function getDatabase(): Knex {
 		return database;
 	}
 
+	if (env.DB_CLIENT === 'mysql') {
+		if (env.DB_CHARSET) {
+			logger.warn(`DB_CHARSET has been deprecated and now has no effect.`);
+			env.DB_CHARSET = null;
+		}
+	}
```

Comment:
```suggestion
```

I think we can still allow DB_CHARSET for power users, as long as we throw appropriate warnings about misusage in the validateDatabaseCharset step below